lib/gds_api/locations_api.rb

```diff
@@ -0,0 +1,30 @@
+require_relative "base"
+require_relative "exceptions"
+
+class GdsApi::LocationsApi < GdsApi::Base
+  # Get a list of local custodian codes for a postcode
+  #
+  # @param [String, nil] postcode The postcode for which the custodian codes are requested
+  #
+  # @return [Array] All local custodian codes for a specific postcode
+  def local_custodian_code_for_postcode(postcode)
+    response = get_json("#{endpoint}/locations?postcode=#{postcode}.json")
+
+    return [] if response.nil?
+
+    response.map { |r| r["local_custodian_code"] }
```

Comment:
We may want to add a `.uniq` at the end of this, to filter out all duplicate custodian codes. (We'd need to add a test for this too - so can make this change in a separate commit).

Sorry I didn't catch this earlier!

It feels a little awkward introducing logic at the GDS API Adapters layer (rather than in Locations API directly) but I think it makes sense here as we only have the [one route](https://github.com/alphagov/locations-api/blob/593e7589c513743bccccd9193f00242bded2accd/config/routes.rb#L6) in Locations API (so no room for that level of granularity), and there's already a precedent for adding logic in GDS API Adapters e.g. in [content_store.rb](https://github.com/alphagov/gds-api-adapters/blob/master/lib/gds_api/content_store.rb).